driver/pparse.mli

```diff
@@ -54,11 +54,12 @@ val apply_rewriters_sig:
 val report_error : error Format_doc.format_printer
 val report_error_doc: error Format_doc.printer
 
+type 'a ast_result = { ast : 'a; source_file : string }
 
 val parse_implementation:
-  tool_name:string -> string -> Parsetree.structure
+  tool_name:string -> string -> Parsetree.structure ast_result
 val parse_interface:
-  tool_name:string -> string -> Parsetree.signature
+  tool_name:string -> string -> Parsetree.signature ast_result
```

Comment:
Those functions are sensitive in term of backward compatibility of the compiler-libs. Moreover, the updated name for ast is already available in `Location.input_name`. Thus I propose to be less principled here, and use this side-band channel for now.